examples/seismic/tti/operators.py

```diff
@@ -35,124 +379,42 @@ def ForwardOperator(model, source, receiver, time_order=2, space_order=4,
-        Gz1r = (ang1 * v.dxr + ang0 * v.dy)
-        Gxx1 = (first_derivative(Gx1p * ang0, dim=x,
-                                 side=right, order=space_order,
-                                 matvec=transpose) -
-                first_derivative(Gx1p * ang1, dim=y,
-                                 side=centered, order=space_order,
-                                 matvec=transpose))
-        Gzz1 = (first_derivative(Gz1r * ang1, dim=x,
-                                 side=right, order=space_order,
-                                 matvec=transpose) +
-                first_derivative(Gz1r * ang0, dim=y,
-                                 side=centered, order=space_order,
-                                 matvec=transpose))
-        Gx2p = (ang0 * u.dx - ang1 * u.dyr)
-        Gz2r = (ang1 * v.dx + ang0 * v.dyr)
-        Gxx2 = (first_derivative(Gx2p * ang0, dim=x,
-                                 side=centered, order=space_order,
-                                 matvec=transpose) -
-                first_derivative(Gx2p * ang1, dim=y,
-                                 side=right, order=space_order,
-                                 matvec=transpose))
-        Gzz2 = (first_derivative(Gz2r * ang1, dim=x,
-                                 side=centered, order=space_order,
-                                 matvec=transpose) +
-                first_derivative(Gz2r * ang0, dim=y,
-                                 side=right, order=space_order,
-                                 matvec=transpose))
-
-        Hp = -(.5 * Gxx1 + .5 * Gxx2)
-        Hzr = -(.5 * Gzz1 + .5 * Gzz2)
-
+    FD_kernel = kernels[(kernel, len(model.shape))]
+    H0, Hz = FD_kernel(u, v, ang0, ang1, ang2, ang3, space_order)
     s = t.spacing
-
+    # Stencils
     stencilp = 1.0 / (2.0 * m + s * damp) * \
         (4.0 * m * u + (s * damp - 2.0 * m) *
-         u.backward + 2.0 * s**2 * (epsilon * Hp + delta * Hzr))
+         u.backward + 2.0 * s ** 2 * (epsilon * H0 + delta * Hz))
     stencilr = 1.0 / (2.0 * m + s * damp) * \
         (4.0 * m * v + (s * damp - 2.0 * m) *
-         v.backward + 2.0 * s**2 * (delta * Hp + Hzr))
-
-    # Add substitutions for spacing (temporal and spatial)
+         v.backward + 2.0 * s ** 2 * (delta * H0 + Hz))
     first_stencil = Eq(u.forward, stencilp)
     second_stencil = Eq(v.forward, stencilr)
     stencils = [first_stencil, second_stencil]
 
+    # SOurce and receivers
     stencils += src.inject(field=u.forward, expr=src * dt * dt / m,
                            offset=model.nbpml)
     stencils += src.inject(field=v.forward, expr=src * dt * dt / m,
                            offset=model.nbpml)
     stencils += rec.interpolate(expr=u + v, offset=model.nbpml)
+    # Add substitutions for spacing (temporal and spatial)
     subs = dict([(t.spacing, dt)] + [(time.spacing, dt)] +
                 [(i.spacing, model.get_spacing()[j]) for i, j
                  in zip(u.indices[1:], range(len(model.shape)))])
+    # Operator
```

Comment:
useless

Reply:
Don't agree, help separating setup parts

Reply:
You can leave blank lines, but fair enough, it's just nitpicking